config/jsonconfig/config.go

```diff
@@ -0,0 +1,109 @@
+/*
+jsonconfig implements configuration, reading json into an ice Module.
+
+To use:
+
+1) Create the Options. List your configurable Interfaces. Each Interface
+can be backed by several named Implementations.
+2) Options.Parse parses bytes and creates a Configuration.
+  a) for each Interface, pick which Implementation.
+  b) json.Unmarshal the json into that Implementation
+  c) Implementation can now be used as a Module or json.Marshal'ed to print its configuration
+3) Configuration is an ice Module that installs all the configured Implementations
+*/
+package jsonconfig
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/scootdev/scoot/ice"
+)
+
+type Options map[string]Interface
+
+type Interface map[string]Implementation
+
+type Implementation interface {
```

Comment:
I'm happy to change it to whatever's best.

Options lets you describe multiple Interfaces, each of which has multiple Implementations. E.g., Cluster is an Interface with multiple Implementations.

So you set up the Options, parse it, and get a Configuration (which is a Module you can install)